src/main/java/org/betonquest/betonquest/VariableInstruction.java

```diff
@@ -29,12 +29,13 @@ public VariableInstruction(final BetonQuestLogger log, final QuestPackage pack,
         if (!instruction.isEmpty() && instruction.charAt(0) != '%' && !instruction.endsWith("%")) {
             throw new IllegalArgumentException("Variable instruction does not start and end with '%' character");
         }
-        super.parts = instruction.substring(1, instruction.length() - 1).split("\\.");
+        final String rawInstruction = instruction.substring(1, instruction.length() - 1);
+        super.data = new Data(rawInstruction, rawInstruction.split("\\."));
     }
 
     @Override
     public String getInstruction() {
```

Comment:
This override is now unnecessary

Reply:
but It will fail some tests in `VariableInstructionTest`, as it needs to be split around dots